File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.swift
import Foundation

func solution(_ sizes: [[Int]]) -> Int {
    var wMaxNumber: Int = 0
    var hMaxNumber: Int = 0
    
    for size in sizes {        
        let w = max(size[0], size[1])
        let h = min(size[0], size[1])        
        wMaxNumber = max(w, wMaxNumber)
        hMaxNumber = max(h, hMaxNumber)
    }
    
    return wMaxNumber * hMaxNumber
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
// 가로든 세로든 더 긴 녀석을 0번 인덱스로 몰아
// 그런 후 제일 큰 가로 * 제일 큰 세로 리턴 때려

function solution(sizes) {    
    let maxW = 0
    let maxH = 0    
    for(const [w,h] of sizes) {
        const big = Math.max(w,h)
        const small = Math.min(w,h)
        maxW = Math.max(maxW, big)
        maxH = Math.max(maxH, small)        
    }        
    return maxW * maxH    
}